cmip6_downscaling/methods/bcsd.py

```diff
@@ -0,0 +1,249 @@
+import os
+
+from skdownscale.pointwise_models import BcAbsolute, BcRelative, PointWiseDownscaler
+
+from cmip6_downscaling.constants import ABSOLUTE_VARS, RELATIVE_VARS
+from cmip6_downscaling.data.cmip import load_cmip
+from cmip6_downscaling.data.observations import get_spatial_anomolies, load_obs
+from cmip6_downscaling.workflows.utils import (
+    load_paths,
+    rechunk_zarr_array,
+    regrid_dataset,
+    write_dataset,
+)
+
+connection_string = os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
+# TODO: add templates for paths, maybe using prefect context. Do this once functions ported to prefect world in bcsd_flow.py
+
+
+def preprocess_bcsd(
+    gcm: str,
+    obs_id: str,
+    train_period_start: str,
+    train_period_end: str,
+    variable: str,
+    coarse_obs_path: str,
+    spatial_anomolies_path: str,
+    connection_string: str,
+    rerun: bool = True,
+) -> tuple[str, str]:
+    """Given GCM and selected obs dataset for a given variable and a given time period, at the coarse scale to match that
```

Comment:
see comment above on a short 1-line description.

Reply:
I don't see an above comment? Are you recommending to shorten the docstring?